libraries/classes/FieldMetadata.php

```diff
@@ -0,0 +1,456 @@
+use const MYSQLI_TYPE_GEOMETRY;
+use const MYSQLI_TYPE_JSON;
+use const MYSQLI_ENUM_FLAG;
+use const MYSQLI_SET_FLAG;
+use function property_exists;
+
+/**
+ * Handles fields Metadata
+ *
+ * NOTE: Getters are not used in all implementations due to the important cost of getters calls
+ */
+final class FieldMetadata
+{
+    public const TYPE_GEOMETRY = 1;
+    public const TYPE_BIT = 2;
+    public const TYPE_JSON = 3;
+    public const TYPE_REAL = 4;
+    public const TYPE_INT = 5;
+    public const TYPE_BLOB = 6;
+    public const TYPE_UNKNOWN = -1;
+    public const TYPE_NULL = 7;
+    public const TYPE_STRING = 8;
+    public const TYPE_DATE = 9;
+    public const TYPE_TIME = 10;
+    public const TYPE_TIMESTAMP = 11;
+    public const TYPE_DATETIME = 12;
+    public const TYPE_YEAR = 13;
+
+    /**
+     * @var bool
+     * @readonly
+     */
+    public $isMultipleKey;
+
+
+    /**
+     * @var bool
+     * @readonly
+     */
+    public $isPrimaryKey;
+
+
+    /**
+     * @var bool
+     * @readonly
+     */
+    public $isUniqueKey;
+
+
+    /**
+     * @var bool
+     * @readonly
+     */
+    public $isNotNull;
+
+
+    /**
+     * @var bool
+     * @readonly
+     */
+    public $isUnsigned;
+
+
+    /**
+     * @var bool
+     * @readonly
+     */
+    public $isZerofill;
+
+
+    /**
+     * @var bool
+     * @readonly
+     */
+    public $isNumeric;
+
+
+    /**
+     * @var bool
+     * @readonly
+     */
+    public $isBlob;
+
+
+    /**
+     * @var bool
+     * @readonly
+     */
+    public $isBinary;
+
+
+    /**
+     * @var bool
+     * @readonly
+     */
+    public $isEnum;
+
+
+    /**
+     * @var bool
+     * @readonly
+     */
+    public $isSet;
+
+    /** @var int|null */
+    private $mappedType;
+
+
+    /**
+     * @var bool
+     * @readonly
+     */
+    public $isMappedTypeBit;
+
+    /**
+     * @var bool
+     * @readonly
+     */
+    public $isMappedTypeGeometry;
+
+    /**
+     * @var bool
+     * @readonly
+     */
+    public $isMappedTypeTimestamp;
+
+    /**
+     * The column name
+     *
+     * @var string
+     */
+    public $name;
+
+    /**
+     * The original column name if an alias did exist
+     *
+     * @var string
+     */
+    public $orgname;
+
+    /**
+     * The table name
+     *
+     * @var string
+     */
+    public $table;
+
+    /**
+     * The original table name
+     *
+     * @var string
+     */
+    public $orgtable;
+
+    /**
+     * The charset number
+     *
+     * @readonly
+     * @var int
+     */
+    public $charsetnr;
+
+    /**
+     * The number of decimals used (for integer fields)
+     *
+     * @readonly
+     * @var int
+     */
+    public $decimals;
+
+    /**
+     * The width of the field, as specified in the table definition.
+     *
+     * @readonly
+     * @var int
+     */
+    public $length;
+
+    /**
+     * A field only used by the Results class
+     *
+     * @var string
+     */
+    public $internalMimeType;
```

Comment:
```suggestion
    public $internalMediaType;
```
See: https://github.com/phpmyadmin/phpmyadmin/issues/15350

Reply:
Good catch, one less occurrence of mime type